engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/inputSettings/ChangeBindingPopup.java

```diff
@@ -64,8 +66,16 @@ public void initialise() {
         bindButton = find("new-binding", UIInputBind.class);
         WidgetUtil.trySubscribe(this, "remove", button -> bindButton.setNewInput(null));
         WidgetUtil.trySubscribe(this, "ok", button -> {
-            bindButton.saveInput();
-            getManager().popScreen();
+            Input newInput = bindButton.getNewInput();
+            currBinds = config.getInput().getBinds();
+            if(currBinds.isBound(newInput) && !newInput.equals(bindButton.getInput())) {
```

Comment:
`if` should be followed by a space. (Set your IDE's code formatter to `docs/formatter.xml` and autoformat so that it's done automatically)